src/unitTest/java/com/serjltt/moshi/adapters/SerializeOnlyJsonAdapterTest.java

```diff
@@ -0,0 +1,61 @@
+package com.serjltt.moshi.adapters;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SerializeOnlyJsonAdapterTest {
+  // Lazy adapters work only within the context of moshi.
+  private final Moshi moshi = new Moshi.Builder()
+      .add(SerializeOnlyJsonAdapter.FACTORY)
+      .add(new Custom.CustomAdapter()) // We need to check that other annotations are not lost.
+      .build();
+
+  @Test public void serializeOnly() throws Exception {
+    JsonAdapter<Data1> adapter = moshi.adapter(Data1.class);
+
+    Data1 fromJson = adapter.fromJson("{\"data\": \"test\"}");
+    assertThat(fromJson.data).isNull();
+
+    fromJson.data = "1234";
+    assertThat(adapter.toJson(fromJson)).isEqualTo("{\"data\":\"1234\"}");
+  }
+
+  @Test public void factoryMaintainsOtherAnnotations() throws Exception {
+    JsonAdapter<Data2> adapter = moshi.adapter(Data2.class);
+
+    Data2 fromJson = adapter.fromJson("{\"data\": \"test\"}");
+    assertThat(fromJson.data).isNull();
+
+    fromJson.data = "1234";
+    String toJson = adapter.toJson(fromJson);
+    assertThat(toJson).isEqualTo("{\"data\":\"1234Custom\"}");
```

Comment:
`fromJson.data` should be `"1234Custom", and the expected json should be `"{\"data\":\"1234\"}"